packages/pigeon/platform_tests/test_plugin/macos/Classes/CoreTests.gen.swift

```diff
@@ -35,6 +35,13 @@ private func wrapError(_ error: Any) -> [Any?] {
   ]
 }
 
+private func nilOrValue(value: Any?) -> Any? {
```

Comment:
The parameter can be just `Any` since it can be inferred. The returned type likely has to be `Any?` since compiler cannot infer the concrete type when you return `nil` in this function

Reply:
This crashes as `Any` since there are instances of optional variables being passed in as `Any?` swift won't allow implicit coercion of `Any?` to `Any`

Reply:
Not sure if I follow the case you described. Can you provide a quick code snippet? 

This is what I just tried and it all worked: 

```
  private func nilOrValue(value: Any) -> Any? {
    if value is NSNull {
      return nil
    }
    return value
  }

    let anOptionalAny: Any? = 3
    let list: [Any] = [NSNull(), 1, "hi", anOptionalAny]
    let bool = nilOrValue(value: list[0]) as! Bool?
    let int = nilOrValue(value: list[1]) as! Int?
    let str = nilOrValue(value: list[2]) as! String?
    let any = nilOrValue(value: list[3]) as! Int?
```